modin/pandas/utils.py

```diff
@@ -104,6 +106,37 @@ def remote(self, *args):
         return result
 
 
+def post_task_gc(func):
+    """Perform garbage collection after the task is executed.
+
+    Usage:
+        ```
+        @ray.remote
+        @post_task_gc
+        def memory_hungry_op():
+            ...
+        ```
+    Note:
+        - This will invoke the GC for the entire process. Expect
+          About 100ms latency.
+        - We have a basic herustic in place to balance of tradeoff between
+          speed and memory. If the task takes more than 500ms to run, we
+          will do the GC.
+    """
+    def wrapped(*args):
+        start_time = time.time()
+
+        result = func(*args)
+
+        duration_s = time.time() - start_time
+        duration_ms = duration_s * 1000
+        if duration_ms > 500:
```

Comment:
Using size of the dataframe could be another good heuristic since the concern for gc is memory. However, I'm fine with using time for now. Thoughts?

Reply:
I think time makes a bit more sense right now because of the relative overhead of GC. We may in the future try to do this another way, but for now this looks good.